### PR TITLE
feat: mssql package

### PR DESCRIPTION
**Pull request by datner**
I was absolutely stumped by the @parameter syntax. I also could not decide how to properly express the `addInputParameter` syntax. Left many artifacts.

Sorry I couldn't get a more functional version, I have never used this cursed sql flavour

### CODE REVIEW

File: packages/mssql/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# @sqlfx/mysql
```
Comment (datner) [inline packages/mssql/CHANGELOG.md:1]: oops 🤦🏻

File: packages/mssql/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+    const compiler = makeCompiler(options.transformQueryNames)
+
+    const transformRows = Client.defaultRowTransform(
+      options.transformResultNames!,
+    )
+
+    const makeConnection = pipe(
+      Effect.acquireRelease(
+        Effect.tap(Effect.sync(() =>
+          new Tedious.Connection({
```
Comment (datner) [inline packages/mssql/src/index.ts:87]: They have may more options, maybe interesting, maybe not.

File: packages/mssql/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+      ),
+      Effect.map((conn): Connection => {
+        const run = (
+          sql: string,
+          _values?: ReadonlyArray<any>,
+          transform = true,
+          rowsAsArray = false,
+        ) =>
+          Effect.async<never, SqlError, any>(resume => {
+            const req = new Tedious.Request(sql, (error, _rowCount, result) => {
```
Comment (datner) [inline packages/mssql/src/index.ts:120]: Very awkward api, the request gets the callback for execution, but isn't the one that executes the query

File: packages/mssql/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+export const makeLayer = (config: Config.Config.Wrap<TediusClientConfig>) =>
+  Layer.scoped(tag, Effect.flatMap(Effect.config(Config.unwrap(config)), make))
+// UPDATE e
+// SET hire_date = t.hire_date
+// FROM dbo.employee e
+// JOIN (
+//     VALUES
+//         ('PMA42628M', '1979-03-15'),
+//         ('PSA89086M', '1988-12-22')
+// ) t (emp_id, hire_date) ON t.emp_id = e.emp_id
```
Comment (datner) [inline packages/mssql/src/index.ts:189]: noise

File: packages/mssql/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+export const makeCompiler = (transform?: (_: string) => string) => {
+  return Statement.makeCompiler<TediousCustom>(
+    (_,v) => `@${v}`,
```
Comment (datner) [inline packages/mssql/src/index.ts:192]: had to change the api because of their name parameter api thing

File: packages/mssql/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+    transform ? _ => escape(transform(_)) : escape,
+    (placeholders, valueAlias, valueColumns, values) => [
+      `(values ${placeholders}) AS ${valueAlias}${valueColumns}`,
```
Comment (datner) [inline packages/mssql/src/index.ts:195]: Just saw that Sql Server has a similar flavor to PostgresQL  so this maybe(??) is valid

File: packages/mssql/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+      values.flat(),
+    ],
+    (type, placeholder) => {
+      switch (type.kind) {
+        case "TediusType": {
+          return [placeholder(), [type.i0 as any, type.i1] as any]
+        }
```
Comment (datner) [inline packages/mssql/src/index.ts:202]: Had more than a couple of experiments trying to make it work

File: packages/mssql/src/index.ts [hunk 8]
```diff
@@ -0,0 +1,403 @@
...
+      }
+    },
+  )
+}
+
+// custom types
+
+type TediousCustom =
```
Comment (datner) [inline packages/mssql/src/index.ts:210]: Not sure where the responsibility should lie, left the types and everything here for your consideration

File: packages/mssql/test/index.test.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+const compiler = _.makeCompiler()
+
+describe("tedious", () => {
+  it("insert helper", () => {
+    const [query, params] = compiler.compile(
+      sql`INSERT INTO people ${sql({ name: "Tim", age: 10 })}`,
+    )
+    expect(query).toEqual(`INSERT INTO people (name,age) VALUES (@name,@age)`)
+    expect(params).toEqual(["Tim", 10])
+  })
```
Comment (datner) [inline packages/mssql/test/index.test.ts:15]: This is probably correct

Comment (datner) [reply packages/mssql/test/index.test.ts:15]: The rest are not guaranteed...

**Review by datner — COMMENTED** [review]
dprint had a field day with a few files

**Review by datner — COMMENTED** [review]
dprint had a field day with a few files

File: packages/mssql/src/index.ts (hunk 6 above)
Comment (tim-smart) [reply packages/mssql/src/index.ts:229]: Still need to tests this

File: packages/mssql/src/index.ts (hunk 8 above)
Comment (tim-smart) [reply packages/mssql/src/index.ts]: Replaced with a lower level `MssqlParam` type.

File: packages/mssql/src/index.ts (hunk 2 above)
Comment (tim-smart) [reply packages/mssql/src/index.ts]: I feel like there are a bunch of missing config items, but we can cross that bridge later.

File: packages/mssql/src/index.ts (hunk 6 above)
Comment (tim-smart) [reply packages/mssql/src/index.ts:311]: Seems to work.

**Pull request merged**